os/arch/arm/src/amebasmart/amebasmart_spi.c

```diff
@@ -1002,8 +1135,66 @@ static void amebasmart_spi_exchange_nodma(FAR struct spi_dev_s *dev,
 			}
 		}
 	}
+#else
+/*************************************************************************
+ * Name: spi_exchange (with DMA)
+ *
+ * Description:
+ *   Exchange a block of data on SPI using DMA
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *   txbuffer - A pointer to the buffer of data to be sent
+ *   rxbuffer - A pointer to a buffer in which to receive data
+ *   nwords   - the length of data to be exchanged in units of words.
+ *              The wordsize is determined by the number of bits-per-word
+ *              selected for the SPI interface.  If nbits <= 8, the data is
+ *              packed into uint8_t's; if nbits >8, the data is packed into uint16_t's
+ *
+ * Returned Value:
+ *   None
+ *
+ ************************************************************************************/
+	uint8_t *rxbuff = NULL;
+	rxbuff = (uint8_t *)rtw_zmalloc(SPI_DMA_MAX_BUFFER_SIZE);
+	if(rxbuff == NULL) {
+		return;
+	}
+	uint8_t *txbuff =  NULL;
+	txbuff = (uint8_t *)rtw_zmalloc(SPI_DMA_MAX_BUFFER_SIZE);
```

Comment:
Regardless of `txbuffer` and `rxbuffer`, do we need to allocate them always?

Reply:
If there is data slicing involved, and also this rtw_zmalloc to ensure 64 bytes aligned.

Reply:
I don't get your point. What I meant is why you always allocate the `rxbuff` and `txbuff` regardless of `rxbuffer` and `txbuffer`. You use them it when rx or tx buffer is given. When `rxbuffer` is null, do you need to allocate the `rxbuff`?

Reply:
I get what you mean, I will make the relevant changes

Reply:
minor comment.
There are similar variables, txbuff, txbuffer and rxbuff, rxbuffer. We can't know what they are different. For code readability, could you rename them to know easily what purposes are?

Reply:
changed to rxbuff_aligned and txbuff_aligned.